tests/test_plugin.c

```diff
@@ -0,0 +1,1425 @@
+    if (rc != SR_ERR_OK) {
+        return 1;
+    }
+
+    // delete all datastore data
+    rc = sr_replace_config(tdata->sess, "plugin", NULL, 0);
+    if (rc != SR_ERR_OK) {
+        return 1;
+    }
+
+    /*
+    *   CANDIDATE
+    */
+    rc = sr_session_switch_ds(tdata->sess, SR_DS_CANDIDATE);
+    if (rc != SR_ERR_OK) {
+        return 1;
+    }
+
+    // reset candidate
+    rc = sr_copy_config(tdata->sess, "plugin", SR_DS_RUNNING, 0);
+    if (rc != SR_ERR_OK) {
+        return 1;
+    }
+
+    return 0;
+}
+
+static void
+store_and_load_example(test_data_t *tdata)
+{
+    int rc;
+    sr_data_t *data = NULL;
+    struct lyd_node *node = NULL;
+    char *str1 = NULL;
+    const char *str2 = "<simple-cont xmlns=\"s\">"
+            "<simple-cont2>"
+            "<ac1>"
+            "<acl1>"
+            "<acs1>a</acs1>"
+            "<acs2>a</acs2>"
+            "</acl1>"
+            "<acl1>"
+            "<acs1>b</acs1>"
+            "<acs3>a</acs3>"
+            "<inner>"
+            "<inner-leaf>a</inner-leaf>"
+            "</inner>"
+            "</acl1>"
+            "</ac1>"
+            "</simple-cont2>"
+            "</simple-cont>";
+
+    rc = lyd_parse_data_mem(tdata->ctx, str2, LYD_XML, LYD_PARSE_STRICT | LYD_PARSE_ONLY, 0, &node);
+    assert_int_equal(rc, LY_SUCCESS);
+
+    rc = sr_replace_config(tdata->sess, "plugin", node, 0);
+    assert_int_equal(rc, SR_ERR_OK);
+
+    // load module
+    rc = sr_get_data(tdata->sess, "/plugin:*", 0, 0, 0, &data);
+    assert_int_equal(rc, SR_ERR_OK);
+
+    rc = lyd_print_mem(&str1, data->tree, LYD_XML, LYD_PRINT_WITHSIBLINGS | LYD_PRINT_WD_ALL_TAG | LYD_PRINT_SHRINK);
+    assert_int_equal(rc, LY_SUCCESS);
+    sr_release_data(data);
+
+    // compare
+    assert_string_equal(str1, str2);
+    free(str1);
+}
+
+/* TEST */
+static void
+test_store_example(void **state)
+{
+    int rc;
+    test_data_t *tdata = *state;
+
+    /* STARTUP */
+    rc = sr_session_switch_ds(tdata->sess, SR_DS_STARTUP);
+    assert_int_equal(rc, SR_ERR_OK);
+    store_and_load_example(tdata);
+
+    /* RUNNING */
+    rc = sr_session_switch_ds(tdata->sess, SR_DS_RUNNING);
+    assert_int_equal(rc, SR_ERR_OK);
+    store_and_load_example(tdata);
+
+    /* CANDIDATE */
+    rc = sr_session_switch_ds(tdata->sess, SR_DS_CANDIDATE);
+    assert_int_equal(rc, SR_ERR_OK);
+    store_and_load_example(tdata);
+}
+
+static void
+store_and_load_complex(test_data_t *tdata)
+{
+    int rc;
+    sr_data_t *data = NULL;
+    struct lyd_node *node = NULL;
+    char *str1 = NULL;
+    const char *str2 =
+            "<simple-cont xmlns=\"s\">"
+            "<simple-cont2>"
+            "<ac1>"
+            "<acl1>"
+            "<acs1>a</acs1>"
+            "<acs2>a</acs2>"
+            "</acl1>"
+            "<acl1>"
+            "<acs1>b</acs1>"
+            "<acs3>a</acs3>"
+            "<inner>"
+            "<inner-leaf>a</inner-leaf>"
+            "</inner>"
+            "</acl1>"
```

Comment:
`uncrustify` does this and makes it unreadable, I can provide a script that formats it better and makes `uncrustify` happy as well.

Reply:
done